core/modules/menu/js/menu.admin.js

```diff
@@ -19,6 +19,60 @@ Backdrop.behaviors.menuAdminFieldsetSummaries = {
   }
 };
 
+Backdrop.behaviors.menu_edit_item_parents = {
+  attach: function (context, settings) {
+    // Get the menu parent options from settings.
+    var menuOptions = settings.menu_edit_item_parents;
+    // Get the current menu name from settings.
+    var menuName = settings.menu_edit_item_menu;
+    // Always move the menu selector before the menu parent selector.
+    $('.form-item-parent-menu').insertBefore('#menu-parent-select-wrapper');
+    // On load set the current menu as default in the menu select.
+    Backdrop.menu_edit_update_parent_list(menuName);
+    // Ensure that the menu parent select list is filtered again when any ajax
+    // process runs.
+    $( document ).ajaxComplete(function( event, xhr, settings ) {
+      if (settings.url=='/system/ajax') {
+        var selected = $('[data-menu-parent] :selected').val().split(':')[0];
+        Backdrop.menu_edit_update_parent_list(selected);
+      }
+    });
+    // On changing the menu select, update the menu parent select.
+    var sel = $('.form-item-parent-menu select');
+    sel.on('change', function () {
+      Backdrop.menu_edit_update_parent_list(this.value);
+    });
+  }
+}
+
+/**
+ * Function to set the options of the menu parent item dropdown.
+ */
+Backdrop.menu_edit_update_parent_list = function (value) {
```

Comment:
Let's group this function under a sub-object for the `menu` module. And use snake case again:

```suggestion
Backdrop.menu = {};

/**
 * Function to set the options of the menu parent item dropdown.
 */
Backdrop.menu.editUpdateParentList = function (value) {
```

Reply:
I don't understand about this, since JQuery is not a favorite language of mine. Also, this function is a copy of the one below it, which doesnt get this treatment, so not sure why its needed here. I could just do what you say here, but best understnd the implications.

Reply:
Yeah, on further consideration, let's just leave it as-is rather than refactoring the whole file.